lib/collection/src/operations/verification/mod.rs

```diff
@@ -199,6 +238,21 @@ pub(crate) fn check_limit_opt<T: PartialOrd + Display>(
     Ok(())
 }
 
+pub(crate) fn check_custom<T: PartialOrd>(
+    value_fn: impl FnOnce() -> Option<T>,
+    limit: Option<T>,
+) -> bool {
+    let Some(limit) = limit else {
+        return true;
+    };
+
+    let Some(value) = value_fn() else {
+        return true;
+    };
+
+    value <= limit
```

Comment:
We _can_ simplify this to:

```rust
value_fn()
    .zip(limit)
    .is_none_or(|(value, limit)| value <= limit)
```

Reply:
Note that the `value_fn()` won't be evaluated lazily anymore then. For our current implementation this doesn't change anything but I think it is nice to have a lazily evaluated function that only checks when strict mode is configured.